### PR TITLE
[Upgrade Tool] Custom JSON Transform API

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

Adds a custom json transform API

- `get`, `set`, `has`, `merge`, `root`

### Why is it needed?

- Simplify json files modifications

### How to test it?

- See the `example in packages/utils/upgrade/resources/transforms/5.0.0`

### CODE REVIEW

File: packages/utils/upgrade/resources/transforms/5.0.0/upgrade-strapi-version.json.ts
Comment (innerdvations) [inline packages/utils/upgrade/resources/transforms/5.0.0/upgrade-strapi-version.json.ts:16]: I would stick with root or source in both places, when I see file.source my first thought is just a raw string of the entire file and I was confused at first about why we were returning a file here and an object down below.

Or honestly something even more explicit, like `parsedJson` `json` `jsonValue` `jsonObject`, not sure, but if you have an idea :)

File: packages/utils/upgrade/src/types.ts
Comment (innerdvations) [inline packages/utils/upgrade/src/types.ts:46]: Can we combine this with the one in @strapi/types so that we don't duplicate it? Maybe it could live in @strapi/types/utils and both this and entity service (and document service) can pull in that

**Review by innerdvations — COMMENTED** [review]
Just a couple minor comments, otherwise looks great

File: packages/utils/upgrade/resources/transforms/5.0.0/upgrade-strapi-version.json.ts
Comment (Convly) [reply packages/utils/upgrade/resources/transforms/5.0.0/upgrade-strapi-version.json.ts]: I agree, it can be confusing. I went with `file.json`

**Review by innerdvations — APPROVED** [review]
LGTM but I didn't QA, waiting until we have most of the pieces merged together and then I'll test it all together

**Pull request merged**
